deployment-template/eks/otel-fargate-container-insights.yaml

```diff
@@ -0,0 +1,552 @@
+        config:
+          global:
+            scrape_interval: 1m
+            scrape_timeout: 40s
+
+          scrape_configs:
+          - job_name: 'kubelets-cadvisor-metrics'
+            sample_limit: 10000
+            scheme: https
+
+            kubernetes_sd_configs:
+            - role: node
+            tls_config:
+              ca_file: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+              insecure_skip_verify: true
+            bearer_token_file: /var/run/secrets/kubernetes.io/serviceaccount/token
+
+            relabel_configs:
+              - action: labelmap
+                regex: __meta_kubernetes_node_label_(.+)
+                # Only for Kubernetes ^1.7.3.
+                # See: https://github.com/prometheus/prometheus/issues/2916
+              - target_label: __address__
+                # Changes the address to Kube API server's default address and port
+                replacement: kubernetes.default.svc:443
+              - source_labels: [__meta_kubernetes_node_name]
+                regex: (.+)
+                target_label: __metrics_path__
+                # Changes the default metrics path to kubelet's proxy cadvdisor metrics endpoint
+                replacement: /api/v1/nodes/$${1}/proxy/metrics/cadvisor
+            metric_relabel_configs:
+              # extract readable container/pod name from id field
+              - action: replace
+                source_labels: [id]
+                regex: '^/machine\.slice/machine-rkt\\x2d([^\\]+)\\.+/([^/]+)\.service$'
+                target_label: rkt_container_name
+                replacement: '$${2}-$${1}'
+              - action: replace
+                source_labels: [id]
+                regex: '^/system\.slice/(.+)\.service$'
+                target_label: systemd_service_name
+                replacement: '$${1}'
+    processors:
+      # rename labels which apply to all metrics and are used in metricstransform/rename processor
+      metricstransform/label_1:
+        transforms:
+          - include: .*
+            match_type: regexp
+            action: update
+            operations:
+              - action: update_label
+                label: name
+                new_label: container_id
+              - action: update_label
+                label: kubernetes_io_hostname
+                new_label: NodeName
+              - action: update_label
+                label: eks_amazonaws_com_compute_type
+                new_label: LaunchType
+
+      # rename container and pod metrics which we care about.
+      # container metrics are renamed to `new_container_*` to differentiate them with unused container metrics
+      metricstransform/rename:
+        transforms:
+          - include: container_spec_cpu_quota
+            new_name: new_container_cpu_limit_raw
+            action: insert
+            match_type: regexp
+            experimental_match_labels: {"container": "(.|\\s)*\\S(.|\\s)*", "LaunchType": "fargate"}
```

Comment:
Since there aren't any anchors, unless the matcher adds implicit start and end anchors (which would be annoying), you can just use `\S`.  That will match if there is any non-whitespace and we don't care about anything else.  It is much simpler and clearer to understand for someone looking at this later.

Reply:
Hi @Aneurysm9! i have tried a lot of new regex like (.|\s)*, (.|\S)*, [\s\S]*, [.\s]*, and [.\S]* but all of them show some errors, do we force need to change this regex now? since we are going to launch this next week and this will not impact on any other code, could we leave this for a phase II change?

Reply:
Did you try simply `\S`?  What errors were produced?

I am concerned about this regex because it appears to be overly complicated and confusing.  It should be simple to resolve and we should resolve it now.

Reply:
yeah i tried it will post some strange error like CrashLoopBackOff if i just use `\S`, and the error shows like cannot retrieve the configuration: error loading config file "/conf/adot-collector-config.yaml": unable to read the file /conf/adot-collector-config.yaml: yaml: line 70: found unknown escape character, but the line 70 in my testing config file is line 49 in the pr, obviously no error on that line, and also the original regex works well, thus got stuck here

Reply:
Since this is in a string in a YAML file representing another YAML file, I don't think you can use `\S` directly, but must escape the `\`, so something like this:

```yaml
experimental_match_labels: {"container": "\\S", "LaunchType": "fargate"}
```

Reply:
We want to make sure it's not null or non empty or non blank. Does it guaranty it has some kind of value?

Reply:
> \S matches any non-whitespace character (equivalent to [^\r\n\t\f\v ])

Reply:
changed to `\\S` and it works